# _Campylobacter_ analysis

_Campylobacter_ are gram-negative bacteria responsible for the majority of the
cases of foodborne bacterial infections ([Kirk et al.
2015](https://journals.plos.org/plosmedicine/article?id=10.1371/journal.pmed.1001921),
[The European Union One Health 2019 Zoonoses
Report](https://www.ecdc.europa.eu/en/publications-data/european-union-one-health-2019-zoonoses-report)).
Although poultry has been pointed as the major source of campylobacteriosis,
several cases have been linked to other sources such as ruminants or environment
([Cody et al.
2019](https://www.eurosurveillance.org/content/10.2807/1560-7917.ES.2019.24.43.1800696)).
Campylobacteriosis causes gastroenteritis, with symptoms that involve diarrhea
and fever, but it may also be responsible for a neurological disorder called
[Guillain-Barré
syndrome](https://en.wikipedia.org/wiki/Guillain%E2%80%93Barr%C3%A9_syndrome).
So far, 17 _Campylobacter_ species and six subspecies have been described, from
which _Campylobacter jejuni_ and _Campylobacter coli_ are the most commonly
associated with human illness ([ECDC
2018](https://www.ecdc.europa.eu/sites/default/files/documents/antimicrobial-consumption-annual-epidemiological-report-2017.pdf)).
Thermophilic _Campylobacter_ species grow at temperatures between 37ºC and 42ºC
(41.5ºC being the optimal temperature) ([Silva et al.
2011](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3180643/pdf/fmicb-02-00200.pdf)).
The thermophilic species are the ones that are of greatest concern for human
illness.

In a WGS approach regarding _Campylobacter_ spp. it might be of importance to
identify the species. The similarity between different isolates (from clinical,
animal or environmental sources), and their respective virulence and
antimicrobial resistance markers is essential for a proper disease surveillance.
_Campylobacter_ serotyping is based on Penner serotyping scheme, which relies on a
hemagglutination assay of lipooligosaccharides (LOS) and of a capsule
polysaccharide (CPS), with CPS being the primary serodeterminant ([Penner &
Hennessy, 2000](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC273687/), [Parkhill
et al. 2000](https://www.nature.com/articles/35001088), [Karlyshev et al.
2000](https://pubmed.ncbi.nlm.nih.gov/10672176/), [Pike et al.
2013](https://journals.plos.org/plosone/article?id=10.1371/journal.pone.0067375#pone.0067375-Parkhill1)).
More than 40 _Campylobacter_ serotypes have been described with this methodology.
Nevertheless, similar to what happens with other species, molecular typing has a
higher discriminatory power, which is useful for epidemiological purposes.

## Typing methods

An ideal typing method presents not only a high discriminatory power, but also
high reproducibility and the possibility of automation. For this reason,
molecular typing is a constantly evolving field always seeking for better
technologies. Nowadays, different techniques can be applied for _Campylobacter_
molecular typing, namely:

- _Pulsed Field Gel Electrophoresis (PFGE)_   
    PFGE is a fragment length restriction analysis that has long been considered
  the most discriminatory typing method for _Campylobacter_ in the pre-WGS era
  ([Sabat et al.
  2013](https://www.eurosurveillance.org/content/10.2807/ese.18.04.20380-en),
  [Frazão et al. 2020](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7203312/)).
  This is currently the “gold-standard” for
  [PulseNet](https://www.cdc.gov/pulsenet/index.html) network, and has been used
  by public health authorities and food regulators for outbreak investigations.

- _MLVA (Multiple locus variable tandem repeat analysis)_   
    Multiple Locus Variable Number of Tandem Repeats Analysis is a PCR-based
  typing method, which is another typing tool used by the
  [PulseNet](https://www.cdc.gov/pulsenet/pathogens/mlva.html) network (before
  WGS). This method is able to differentiate fast-evolving bacteria even if they
  look similar with PFGE. Therefore, MLVA is usually performed as a complement
  to PFGE results, thus providing a useful resource during outbreaks
  ([Techaruvichit et al. 2015](https://pubmed.ncbi.nlm.nih.gov/26025899/)).

- _MLST (Multi-Locus Sequence Typing)_   
    As for other bacteria, a MLST method based on 7-locus (asp, gnl, glt, gly,
  pgm, tkt, and unc) has been developed for _Campylobacter_ ([Dingle et al.
  2001](https://pubmed.ncbi.nlm.nih.gov/11136741/)). MLST can provide faster
  results compared to PFGE, and it is highly reproducible. However, it shows
  lower discriminatory power than PFGE and MLVA ([Sabat et al.
  2013](https://www.eurosurveillance.org/content/10.2807/ese.18.04.20380-en),
  [Techaruvichit et al. 2015](https://pubmed.ncbi.nlm.nih.gov/26025899/),
  [Frazão et al. 2020](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7203312/)),
  and therefore it was suggested that it should be used as a complement to PFGE
  ([Frazão et al. 2020](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7203312/)).
  A big advantage of MLST analysis for _Campylobacter_ in comparison for instance
  to PFGE, is the existence of a curated database with common nomenclature which
  allows the comparison of results between studies
  ([PubMLST](http://pubmlst.org/campylobacter/)),
  which has made this technique being widely used in epidemiological studies.

- _Sequencing of the short variable region (SVR) of the flaA gene_  
    This method relies on the analysis of the genetic sequence of flaA in
  comparison with the alleles present in
  [PubMLST](https://pubmlst.org/organisms/campylobacter-jejunicoli), and has
  been described as a fast, discriminatory and reproducible tool to discriminate
  among _Campylobacter_ isolates. This technique is useful in combination with
  PFGE or MLST to differentiate outbreak-related isolates ([Niederer et al.
  2012](https://aem.asm.org/content/78/1/288), [Mohan & Habib,
  2019](https://bmcresnotes.biomedcentral.com/articles/10.1186/s13104-019-4107-5),
  [Frazão et al. 2020](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7203312/)).
  Nevertheless, PFGE and MLVA are more discriminatory and used tools for
  _Campylobacter_ typing in epidemiology ([Frazão et al.
  2020](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7203312/)).

- _CRISPR_  
    High-resolution DNA melt curve analysis (HRMA) of the CRISPR region can be
  used to differentiate among _Campylobacter_ isolates ([Price et al.
  2007](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC1907115/#fn1)). This
  technique has been shown to be less discriminatory than PFGE, MLST or even SVR
  of the flaA gene ([Frazão et al.
  2020](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7203312/)). Nevertheless,
  when used in combination with another typing method such as MLST, this
  technique has proven to be useful for epidemiological studies ([Kovanen et al.
  2014](https://sfamjournals.onlinelibrary.wiley.com/doi/full/10.1111/jam.12503)).

- _WGS (Whole-Genome Sequencing)_  
    With the advent of NGS technologies, WGS was proven to be useful for
  _Campylobacter_ outbreak investigation ([Joensen et al.
  2020](https://wwwnc.cdc.gov/eid/article/26/3/19-0947_article)). The
  _Campylobacter_ genome size is approximately 1.8Mb with ~1,800 genes. By
  providing information at the genomic level, WGS allows not only a highly
  discriminatory typing (cgMLST, wgMLST and SNP-typing), but also to establish
  the backward compatibility with previously mentioned molecular typing methods,
  as 7-loci MLST, which, for this reason, will tend to continue to be used.
  Furthermore, it allows the analysis of specific genes, such as virulence
  factors and antimicrobial resistance genes, contributing to a better
  understanding of the different pathogenic populations. Genetic clustering
  using WGS can be performed on any distance measure (eg. issued from allelic
  differences detected using cgMLST typing) or evolutionary-model based
  clustering (ie. phylogenetics) relying on variants/SNPs detection.
  [PulseNet](https://www.cdc.gov/pulsenet/pathogens/wgs.html) network is making
  efforts to implement WGS as a routine tool to replace PFGE and MLVA.
  Nevertheless, this is still not the routine in the case of _Campylobacter_. 

## “One Health” surveillance and WGS of _Campylobacter_

The identification of  sources of infection and the knowledge of pathogens’
genomic features is essential for proper surveillance and outbreak
monitorization. Hence, an integrated analysis of clinical, food and veterinary
samples relying on the concept of One Health is the key to achieve a good
surveillance system. As shown
[here](https://www.cdc.gov/listeria/surveillance/whole-genome-sequencing.html)
by PulseNet network, the high discriminatory power of WGS increases the chances
to find the bacterial source of infection, and possibly reduces the time that it
takes. Indeed, WGS analysis has proven to be an effective way to determine the
genetic clustering of _Campylobacter_ isolates, as well as the source of
infections ([Joensen et al.
2020](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7045838/)). According to the
[European Union One Health 2019 Zoonoses
report](https://www.efsa.europa.eu/en/efsajournal/pub/6406), surveillance
systems for infections by _Campylobacter_ are present in almost all member states,
with the notification of campylobacteriosis being mandatory in 21 countries.
Moreover, _Campylobacter_ is monitored along the food chain. Nevertheless, WGS is
not yet being implemented in routine _Campylobacter_ surveillance.

## WGS lab protocol

### DNA extraction

Before DNA extraction, _Campylobacter_ is cultured in the laboratory. These
bacteria are [microaerophilic](https://en.wikipedia.org/wiki/Microaerophile),
and for this reason they should be cultured under an oxygen-reduced atmosphere
([Buss et al.
2019](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6520473/#CR23)). Moreover, _C.
jejuni_ is usually cultured at 41.5ºC, as it only grows at temperatures between
30ºC and 42ºC ([Duffy & Dykes,
2006](https://sfamjournals.onlinelibrary.wiley.com/doi/full/10.1111/j.1472-765X.2006.02019.x)).
Regarding DNA extraction, there is not a standard protocol or kit that is used,
but many studies use QIAGEN DNeasy Blood or Tissue kit or DNA QIAamp Mini Kit
(Qiagen, The Netherlands) ([Meistere et al.
2019](https://www.eurosurveillance.org/content/10.2807/1560-7917.ES.2019.24.31.1800357),
[Dunn et al.
2018](https://www.microbiologyresearch.org/content/journal/mgen/10.1099/mgen.0.000227),
[Dahl et al.
2020](https://link.springer.com/article/10.1007%2Fs10096-020-04043-y), [Joensen
et al. 2020](https://wwwnc.cdc.gov/eid/article/26/3/19-0947_article)).

### Sequencing technology

There is not a prefered WGS technology to sequence _Campylobacter_. Similar to
other fields, Illumina paired-end reads represent the most commonly used
strategy. Due to the number of samples that can be handled at a single run and
the possible higher read size, MiSeq sequencing machines seem to be the choice
for the majority of the labs.

## Bioinformatics protocol

### Mapping or assembly

The first step to perform when receiving the sequencing data, is to evaluate
the sequencing quality and perform trimming and cleaning of the reads (see [Data
preprocessing](../Pipelines/data_preprocessing.md)). 

The cleaned sequence data can then be used for downstream analysis following one
of two approaches (or both in parallel, check [Data
production][../Pipelines/data_production.md]): 
- _De novo_ genome assembly of the sample(s),
- Read mapping of each sample on a reference sequence (obtained from a database
  or by _de novo_ genome assembly of one of your samples)

It is important to note that both approaches have advantages and disadvantages.
The decision on which of them to follow should be made according to what is more
appropriate for the data at hand, and the purpose of the analyses. _De novo_
genome assembly of all sequenced isolates followed by their annotation seems to
be a common approach in studies including _Campylobacter_ genomes, which then
perform a cgMLST analysis. A commonly used _de novo_ genome assembler for
_Campylobacter_ is [SPAdes](https://cab.spbu.ru/software/spades/) ([Dunn et al.
2018](https://www.microbiologyresearch.org/content/journal/mgen/10.1099/mgen.0.000227),
[Redondo et al.
2019](https://journals.plos.org/plosone/article?id=10.1371/journal.pone.0219088),
[Kelley et al.
2020](https://www.frontiersin.org/articles/10.3389/fmicb.2020.571064/full)). It
performs very well and is freely available. As for read mapping, when performed,
it usually relies on the usage of
[Bowtie](http://bowtie-bio.sourceforge.net/bowtie2/manual.shtml) or
[BWA](http://bio-bwa.sourceforge.net) ([Golz et al.
2020](https://www.nature.com/articles/s41598-020-60320-y), [Dunn et al.
2018](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6249439/), [Mandal et al.
2017](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5557513/), [Wallace et al.
2020](https://journals.plos.org/plosone/article?id=10.1371/journal.pone.0236889),
[Chung et al. 2016](https://www.nature.com/articles/srep38442)). There are
command-line pipelines, such as [INNUca](https://github.com/B-UMMI/INNUca), which incorporate these programs and
provide the opportunity to automatically perform all the analyses from quality
control to genome assembly. If a platform with predefined pipelines (and that
usually does not require bioinformatics skills) is preferred,
[Enterobase](https://enterobase.readthedocs.io/en/latest/) is available for
_Campylobacter_. In addition, the [IRIDA](https://irida.ca/) system and [CLC
Genomics
Workbench](https://digitalinsights.qiagen.com/products-overview/discovery-insights-portfolio/analysis-and-visualization/qiagen-clc-genomics-workbench/)
is in common use. 

### Choosing a reference genome

Should an analysis require the use of a reference genome, the choice of the
reference genome is a crucial step. Analyses relying on read-mapping approaches
might be strongly influenced by reference choice, as the genetic distance
between  the reference and the sample may influence the performance of
downstream steps, namely SNPs/INDELs calling ([Pightling et al.
2014](https://journals.plos.org/plosone/article?id=10.1371/journal.pone.0104579),
[Pightling et al.
2015](https://bmcresnotes.biomedcentral.com/articles/10.1186/s13104-015-1689-4)).
This reference can be picked from the samples (after genome assembly), or from a
public database. A read mapping approach is not commonly used in _Campylobacter_
analysis, and for this reason there is not a specific reference genome in public
databases that is in common use. 

### Getting SNPs

How to detect SNPs is described earlier. 

Briefly, there are three different approaches.

- Perform _de novo_ genome assembly of each sample and then align their genomic
  sequences (or gene sequences after annotation). _Campylobacter_ analyses usually
  use [MAUVE](http://darlinglab.org/mauve/mauve.html) or
  [PRANK](https://www.ebi.ac.uk/research/goldman/software/prank) to align the
  genomes ([Clark et al.
  2018](https://journals.plos.org/plosone/article?id=10.1371/journal.pone.0190836),
  [Weis et al. 2016](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5118927/),
  [Fiedoruk et al.
  2019](https://gutpathogens.biomedcentral.com/articles/10.1186/s13099-019-0313-x),
  [Parker et al.
  2021](https://www.frontiersin.org/articles/10.3389/fcimb.2021.607747/full)).
  The last aligner is mostly used as part of the pan-genome pipeline
  [Roary](https://sanger-pathogens.github.io/Roary/).

- Use a reference genome where the reads of all the samples will be mapped
  (check above), and then use a variant-calling pipeline to determine the
  polymorphic positions.  [CFSAN
  SNP](https://snp-pipeline.readthedocs.io/en/latest/) is a commonly used
  pipeline which performs both processes (read mapping and variant calling).
  [Snippy](https://github.com/tseemann/snippy) is also a commonly used
  alternative.

- Determine the polymorphic positions in the sample by analyzing the k-mer
  pattern using
  [kSNP](https://academic.oup.com/bioinformatics/article/31/17/2877/183216). For
  this approach you can either provide the genome assembly, or the cleaned
  genomic reads. This is the less frequently used approach for _Campylobacter_.

Each of these approaches provides you with information about the genetic
variability of your dataset. This information can then be used to perform
SNP-based clustering and phylogenetic analysis. Alternatively, if you follow a
read mapping approach, you can replace the reference nucleotide by the observed
allele, and consequently reconstruct the haplotype of each sample. This is the
approach used by the [CFSAN SNP](https://snp-pipeline.readthedocs.io/en/latest/)
pipeline.

### Getting alleles and allele differences

The allele sequences of the samples can be retrieved by:

- Replacing the nucleotide of the reference genome by the observed alternative
  allele, and then retrieve the sequence of each gene of interest considering
  the genome annotation of the reference.
- Obtaining the _de novo_ genome assembly of each sample, and performing the
  respective genome annotation. [Prokka](https://github.com/tseemann/prokka) is
  acommonly used program for _Campylobacter_.
- Some allele callers, such as [chewBBACA](https://github.com/B-UMMI/chewBBACA),
  provide locus-specific alignments in an automated manner, being a good option
  to determine the allelic profile of samples. 

It is important to note that nowadays there are several platforms which can
automatically do all this analysis. One of the more commonly used for _Campylobacter_ is
[BIGSdb](https://bigsdb.readthedocs.io/en/latest/). These platforms provide
assembly, serotyping and allele calling. Several of these platforms are
mentioned in the [xMLST](../Clustering/xMLST.md) section.


### Allele based typing

Allele-based typing consists of retrieving clustering information considering
the different alleles present in a population for a given set of genes (e.g. the
core genome). With the advent of WGS, the 7-loci based MLST approach was broadened to the use
of a cgMLST or a wgMLST approach. In this context, there is a public cgMLST
scheme which has been used in _Campylobacter jejuni/coli_ analysis considering an
allele-based approach. This scheme comprises [1,343
loci](https://pubmlst.org/bigsdb?db=pubmlst_campylobacter_seqdef&page=schemeInfo&scheme_id=4)
([Cody et al. 2017](https://jcm.asm.org/content/55/7/2086)).

Platforms available for cgMLST typing of _Campylobacter_ include
[BIGSdb](https://bigsdb.readthedocs.io/en/latest/),
[BioNumerics](http://www.applied-maths.com/sites/default/files/extra/Release-Note-Salmonella-enterica-schema.pdf),
[IRIDA](https://www.irida.ca/platform-overview-2/), [Pathogen
Watch](https://pathogen.watch), and [Ridom
SeqSphere+](https://www.ridom.de/seqsphere/). BIGSdb and BioNumerics seem to be
commonly used by the community. 


### SNP based typing

A SNP-based approach relies on the comparison of SNPs in a population. This
strategy can be seen as an alternative to the allele-based approach, but many
studies actually perform both of them and assess the overlap of the results. For
a SNP-based analysis all of the the SNPs that are present in the samples need to
be acquired and used to obtain clustering information. Examples of publicly
available pipelines for SNP-based typing are: 

- Center for Food Safety and Applied Nutrition [(CFSAN)](https://snp-pipeline.readthedocs.io/en/latest/) HqSNPs pipeline
- [Lyve-SET](https://github.com/lskatz/Lyve-SET) pipeline for HqSNPs typing 
- [SNV-Phyl](https://snvphyl.readthedocs.io/en/latest/) (Canadian Public Health Agency)  
- [PHEnix](https://github.com/phe-bioinformatics/PHEnix) (The Public Health England SNP calling pipeline)

### Outbreak definition

As defined by the [World Health Organization](http://www.emro.who.int/health-topics/disease-outbreaks/index.html), “a disease outbreak is the
occurrence of cases of disease in excess of what would normally be expected in a
defined community, geographical area or season”. WGS data provides a high
discriminatory power allowing clustering of different isolates (from different
geographical areas, and clinical, animal or environmental sources) according to
their genomic similarity. This contributes not only to an earlier detection of
outbreaks and determination of contamination sources, but also to the detection
of more outbreaks, as has been reported by
[PulseNet](https://www.cdc.gov/listeria/surveillance/whole-genome-sequencing.html)
network for _Listeria_. It is difficult to establish a clear cluster outbreak
definition, a threshold at which we decide whether two isolates belong to the
same genetic cluster, thus linking two cases of infection. Epidemiological
related _Campylobacter_ isolates can be distinguished from unrelated ones
([Llarena et al. 2017](https://jcm.asm.org/content/55/5/1269)). Nevertheless,
the genomic variability within an outbreak-related clade varies depending not
only on the dataset, but also on the methodology used (e.g. which MLST or cgMLST
scheme is used). Furthermore, mixed infections may also influence the results
([Llarena et al. 2017](https://jcm.asm.org/content/55/5/1269)). In two
independent outbreaks, a 3 SNPs variation has been found among the isolates
([Revez et al.
2014a](https://jcm.asm.org/content/52/8/2782?ijkey=d47b6f6b09627d48d93b7ddcecc9463909c03601&keytype2=tf_ipsecsha)
and [Revez et al. 2014b](https://pubmed.ncbi.nlm.nih.gov/25196593/)). Using a
732-core-gene schema, [Clark et al.
(2016)](https://bmcgenomics.biomedcentral.com/articles/10.1186/s12864-016-3340-8)
found 4 allele differences between isolates. [Lahti et al.
(2017)](https://pubmed.ncbi.nlm.nih.gov/27334628/) described a maximum of 1
allele difference between clinical isolates, considering a reference-based
cgMLST with 1,271 loci. Therefore, so far, there is no specific threshold used
to define _Campylobacter_ outbreaks, and more studies on the genetic variation
within and between _Campylobacter_ populations would provide a great contribution
to the field.

### Virulence and AMR

Similar to other pathogens, several genes are important for _Campylobacter_
ability to cause infection, and therefore genes such as cadF and ciaB have been
described as medically relevant (eg. [Wu et al.
2016](https://lib.dr.iastate.edu/cgi/viewcontent.cgi?article=1138&context=vmpm_pubs),
[Dasti et al. 2009](https://pubmed.ncbi.nlm.nih.gov/19665925/), [Fiedoruk et al.
2019](https://gutpathogens.biomedcentral.com/articles/10.1186/s13099-019-0313-x),
[Chukwu et al. 2019](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6617328/)).
Moreover, despite the majority of infections not requiring the administration of
antimicrobial drugs, in severe cases of disease antimicrobial therapy can be
provided. In recent years, an increased resistance to these drugs has been
observed in _Campylobacter_ becoming a concern for public health authorities
([CDC](https://www.cdc.gov/campylobacter/campy-antibiotic-resistance.html)).

Several studies have determined genes and respective variations which are
potentially related with increased virulence or specific antimicrobial
resistance (e.g. [Bravo et al.
2021](https://journals.plos.org/plosntds/article?id=10.1371/journal.pntd.0009207),
[Lluque et al.
2017](https://www.hindawi.com/journals/scientifica/2017/7848926/), [Gahamanyi et
al.
2021](https://www.frontiersin.org/articles/10.3389/fmicb.2021.622275/full?&utm_source=Email_to_authors_&utm_medium=Email&utm_content=T1_11.5e1_author&utm_campaign=Email_publication&field=&journalName=Frontiers_in_Microbiology&id=622275#B23),
[Aksomaitiene et al. 2021](https://pubmed.ncbi.nlm.nih.gov/33383765/)).
Moreover, the existence of several events of horizontal gene transfer may
contribute to increase the list of relevant genes for this species
([Aksomaitiene et al. 2021](https://pubmed.ncbi.nlm.nih.gov/33383765/), [Hull et
al.
2021](https://journals.plos.org/plosone/article?id=10.1371/journal.pone.0246571)).
For this reason, it is important to determine the presence of medically
important genes/variations in the isolates. As mentioned in the [“Virulome and
Resistome" section](../Typing/virulence_amr.md), where more details can be found, this is performed by
comparing the genome to a database comprising a set of genes of interest.
Examples of predefined resistome databases are mentioned in the same section. 
