# _Listeria monocytogenes_ analysis

The gram-positive bacterium _Listeria monocytogenes_ is the causative agent of
[Listeriosis](https://www.efsa.europa.eu/en/topics/topic/listeria), a foodborne
disease (reviewed in [Buchanan et al.
2017](https://www.sciencedirect.com/science/article/pii/S0956713516306892)).
Listeriosis can be particularly severe, potentially deadly, in elderly and
immunocompromised patients. It can cause misscariage in pregnant women or
stillbirth. Delay between exposure and illness, and the possibility of
consumption of contaminated products spread over-time (eg. frozen products)
represent a challenge to the identification of epidemiological links between
infection cases (eg. [Datta and Burall
2018](https://www.sciencedirect.com/science/article/abs/pii/S0740002017303118)).

[Matle et al. (2020)](http://www.scielo.org.za/pdf/ojvr/v87n1/13.pdf) reviewed
the different aspects of Listeriosis. They provide an overview of known _L.
monocytogenes_ virulence factors, as well as diagnostics and treatment options.
The state-of-the-art dry-lab approaches employed to the study of _L.
monocytogenes_ are described in [Luth et al.
(2018)](https://www.sciencedirect.com/science/article/pii/S092422441730540X).

Four evolutionary lineages have been identified in _L. monocytogenes_. These
bacteria can be found in a variety of hosts and in environmental samples
[(Buchanan et al.
2017)](https://www.sciencedirect.com/science/article/pii/S0956713516306892). _L.
monocytogenes_ genome is approximately 3Mb, with approximately 2,900 genes ([Den
Bakker et al.
2010](https://bmcgenomics.biomedcentral.com/articles/10.1186/1471-2164-11-688)).
At least 13 serotypes have been identified (see Figure 1 in [Ragon et al.
2008](https://journals.plos.org/plospathogens/article?id=10.1371/journal.ppat.1000146)).
Early studies showed that despite this diversity, the majority of human
infections are caused by isolates belonging to serotypes 1/2a, 1/2b  and 4b (eg.
[Burall et al.
2017](https://journals.plos.org/plosone/article?id=10.1371/journal.pone.0176912)).
Serotyping is based on an antigen-antibody reaction using somatic (O) and
flagellar (H) antigens. Although serotyping has traditionally been used to
characterize _L. monocytogenes_ isolates, it has gradually been replaced by
molecular typing methods that provide enhanced discriminatory power and
therefore represent a more suitable approach for epidemiological studies (see
eg. [Datta and Burall
2018](https://www.sciencedirect.com/science/article/abs/pii/S0740002017303118)(not
OA), for an overview of the methods that have been employed for _L. monocytogenes_
analyses).

## Typing methods

_L. monocytogenes_ molecular typing is a constantly evolving field. An ideal
typing method presents not only a high discriminatory power, but also high
reproducibility and the possibility of automation. Nowadays, different
techniques can be applied for _L. monocytogenes_ molecular typing, namely:

- _Pulsed Field Gel Electrophoresis (PFGE)_   
    PFGE is a fragment length restriction analysis ([Dalmasso et al.
  2014](https://pubmed.ncbi.nlm.nih.gov/24792548/)) that has long been considered 
  as the “gold-standard” for _L. monocytogenes_ typing due to its high
  discriminatory power in the pre-WGS era. This method has been used by
  [PulseNet](https://www.cdc.gov/pulsenet/index.html) to connect cases of
  disease through the comparison of their DNA fingerprints, and consequently
  identify potential outbreaks. Despite its robustness, PFGE is time-consuming,
  difficult to standardize ([Van Walle
  2018](https://www.eurosurveillance.org/content/10.2807/1560-7917.ES.2018.23.33.1700798))
  and lacks discriminatory power for outbreak delineation. Nevertheless, despite
  these disadvantages, PFGE still represented the best compromise between time
  and discriminatory power in _L. monocytogenes_ molecular typing until the advent
  of NGS technologies. It was thus being considered as the “gold-standard”
  typing method for _L. monocytogenes_ for many years, and played an important
  role in _L. monocytogenes_ surveillance and in the resolution of multiple
  outbreaks.


- _MLVA (Multiple locus variable tandem repeat analysis)_  
    Given the drawbacks of PFGE, other typing methods started being considered as
  good alternatives or at least complements to PFGE analysis. MLVA represents
  another method of DNA fingerprinting. This method has the advantage of
  detecting fast-evolving bacterial strains among isolates which may look the
  same with PFGE. However, it requires highly trained technicians and does not
  have a standardized protocol for multiple pathogens. This is why it is not
  used as a routine typing method, but rather as a complementary method to PFGE
  by [PulseNet](https://www.cdc.gov/pulsenet/index.html) for some
  microorganisms, but not _L. monocytogenes_. For this reason, it does not
  represent a standard method for surveillance of _Listeria_, but it is used by
  the scientific community to explore the diversity of these bacteria (e.g.
  [Saleh-Lakha et al. 2013](https://aem.asm.org/content/79/20/6472), [Lunestad
  et al. 2013](https://pubmed.ncbi.nlm.nih.gov/23218175/)). [Chenal-Francisque
  et al. (2013)](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3716049/) compares
  MLVA performance to PFGE and MLST. 


- _Multiplex-PCR for classifying 5 serogroups_  
    Consisting of the amplification of 5 different genes (lmo0737, lmo1118,
  ORF2110, ORF2819 and prs), this method was developed in order to facilitate
  serotyping discrimination by quickly classifying _L. monocytogenes_ into 5
  serogroups ([Borucki and Call 2003](https://jcm.asm.org/content/41/12/5537),
  [Doumith et al.
  2004](https://journals.asm.org/doi/10.1128/JCM.42.8.3819-3822.2004), [Matle et
  al. 2020](http://www.scielo.org.za/pdf/ojvr/v87n1/13.pdf)). Nevertheless,
  despite being a quick method to implement, it has low discriminatory power,
  which makes it less suitable for outbreak detection and investigation.


- _MLST (Multi-Locus Sequence Tying)_  
    DNA sequencing allows unambiguous identification of genetic differences by
  direct comparison of allele sequences between samples, and sequencing
  information can be easily shared between laboratories. Therefore, DNA
  sequencing provides a robust solution for molecular typing. In this context, a
  7-gene MLST (housekeeping genes) method was developed for _L. monocytogenes_
  ([Salcedo et al. 2003](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC149676/),
  [Matle et al. 2020](http://www.scielo.org.za/pdf/ojvr/v87n1/13.pdf)).
  Sequence types (ST) represent unique combinations of the MLST alleles. Clonal
  complexes (CC) are groups of ST differing by no more than one allele to
  another isolate belonging to the same CC ([Ragon et al.
  2008](https://journals.plos.org/plospathogens/article?id=10.1371/journal.ppat.1000146),
  [Henri et al. 2016](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5007763/)). A
  significant drawback of this method is that it requires multiple PCR reactions
  which cannot be multiplexed.


- _Ribosomal multi-locus typing (rMLST)_  
    rMLST has also been employed for strain characterisation ([Jolley et al.
  2012](https://www.microbiologyresearch.org/content/journal/micro/10.1099/mic.0.055459-0)).
  This typing method has recently been employed for WGS data quality control of
  _L. monocytogenes_ to detect potential intra-species contamination (admixture)
  of sequencing data ([Low et al. 2019](https://peerj.com/articles/6995/)).


- _MVLST (Multi-virulence-locus sequence typing)_  
    Similar to MLST, but considering a set of virulence (prfA, inlB, and inlC) and
  virulence-related genes (dal, lisR, and clpP), which has been shown to
  accurately differentiate epidemic clones (see [Lomonaco et al.
  2013](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3558006/), [Cantinelli et
  al. 2013](https://jcm.asm.org/content/jcm/51/11/3770.full.pdf), [Burall et al.
  2017b](https://www.frontiersin.org/articles/10.3389/fpubh.2017.00241/full)).
  
  
- _WGS (Whole-Genome Sequencing)_   
    With the advent of NGS technologies, WGS technology has led to the improvement
  of small listeriosis outbreak investigation and is currently being regarded as
  the new “gold-standard” in the analysis of _L. monocytogenes_ ([Nadon et al.
  2017](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5479977/pdf/eurosurv-22-30544.pdf)).
  By providing information at the genomic level, WGS allows not only a highly
  discriminatory typing (cgMLST, wgMLST and SNP-typing), but also to establish
  backward compatibility with previously mentioned molecular typing methods, such as
  the 7-genes MLST, Multiplex-PCR, rMLST and MVLST, which, for this reason, will
  tend to continue to be used. Furthermore, it allows the analysis of specific
  genes, such as virulence factors and antimicrobial resistance genes. Genetic
  clustering using WGS can be performed on any distance measure (eg. issued from
  allelic differences detected using cgMLST typing) or evolutionary-model based
  clustering (ie. phylogenetics) relying on variants/SNPs detection.
  [PulseNet](https://www.cdc.gov/pulsenet/pathogens/wgs.html) has been
  implementing WGS for _Listeria_ surveillance and outbreak monitoring. Their
  results have shown that [using WGS increases the number of outbreaks detected, and
  earlier outbreak detection facilitates timely action, thus
  limiting the extent of outbreaks](https://www.cdc.gov/listeria/surveillance/whole-genome-sequencing.html).
  Similar studies in the EU have confirmed those findings ([Nielsen et al.
  2017](https://efsa.onlinelibrary.wiley.com/doi/10.2903/sp.efsa.2017.EN-1151),
  [Van Walle et al. 2018](https://pubmed.ncbi.nlm.nih.gov/30131096/), [Moura et
  al. 2016](https://pubmed.ncbi.nlm.nih.gov/27723724/)). For this reason,
  efforts have been made in order to make WGS widely used for _Listeria_
  surveillance, replacing PFGE and serotyping methods. In a near future,
  WGS-based _Listeria_ surveillance is expected to be implemented in most
  developed countries.

## “One Health” surveillance and WGS of _L. monocytogenes_

The identification of infection sources is essential for outbreak resolution.
Hence, an integrated analysis of clinical, food and veterinary samples relying
on the concept of One Health is the key to achieve a good surveillance system.
As shown
[here](https://www.cdc.gov/listeria/surveillance/whole-genome-sequencing.html)
by PulseNet network, the high discriminatory power of WGS increases the chances
to find the source of infection, and possibly reduces the time that it takes to
identify the source. Indeed, as reported by the
[WHO](https://apps.who.int/iris/bitstream/handle/10665/272430/9789241513869-eng.pdf?sequence=1),
the use of WGS on _Listeria_ strains has resulted in more accurate detection of
clusters and allowed more outbreaks to be successfully resolved. However,
several factors are hindering the implementation of a generalized WGS-based
surveillance system. For instance, while the notification of clinical cases of
listeriosis is mandatory in most EU members, most of the monitoring data on _L.
monocytogenes_ in animals and food are generated by non-harmonised monitoring
schemes across member states and for which mandatory reporting requirements do
not exist
([ECDC 2019](https://www.ecdc.europa.eu/sites/default/files/documents/zoonoses-EU-one-health-2019-report.pdf)).
Moreover, WGS requires laboratories to be equipped with expensive technologies,
and highly skilled technicians able to analyze the data, which is not affordable
for many countries. Furthermore, the resources for implementation of WGS differ
between different sectors (human health, animal health and food safety), thus
complicating the implementation of “One health” surveillance. Despite all these
issues in the implementation of a proper WGS-based system, as _L. monocytogenes_
was the selected bacteria to start implementing WGS at an European level, it is
already some steps ahead from other pathogenic agents regarding WGS-based
surveillance. Indeed, _L. monocytogenes_ is currently the most frequently
WGS-based typed pathogen for surveillance and outbreak investigation in all
sectors in Europe ([ECDC et al.
2019](https://efsa.onlinelibrary.wiley.com/doi/epdf/10.2903/sp.efsa.2019.EN-1337)),
and according to the [ECDC
roadmap](https://www.ecdc.europa.eu/en/publications-data/ecdc-roadmap-integration-molecular-typing-and-genomic-typing-european-level),
the capacity of the member states for use of WGS as a complement or replacement 
technology for PFGE is already significant and progressing fast. 

## WGS lab protocol

### DNA extraction

Before DNA extraction, _L. monocytogenes_ is cultured in the laboratory (usually
liquid media). For proper growth, these bacteria need a medium containing the
seven amino acids for which they are
[auxotrophic](https://en.wikipedia.org/wiki/Auxotrophy) (arginine, cysteine,
glutamine, isoleucine, leucine, methionine, and valine) and four additional
vitamins (biotin, riboflavin, thiamine, and thioctic acid). Brain Heart Infusion
(BHI) is a nutrient-rich medium harboring all these ingredients, thus being the
most commonly used medium for _Listeria_ culture (check [Jones and D’Orazio
2013](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3920655/) for more details).
An overnight incubation at 37 ºC in BHI is usually performed before DNA
extraction. Regarding DNA extraction, there is no standard methodology or kit
used for _L. monocytogenes_. However, commonly used kits include DNeasy Blood and
Tissue kit (Qiagen) or Wizard Genomic purification kit (Promega). 

### Sequencing technology

There is not a prefered WGS technology to sequence _L. monocytogenes_. Similar to
other fields, Illumina paired-end reads represent the most commonly used
strategy. Due to the number of samples that can be handled at a single run and
the possible higher read size, MiSeq sequencing machines seem to be the choice
for many labs. Long-read sequencing technologies are now becoming more
frequently used (alone and/or in combination with short-read sequencing). This
because of the improvements in the error rates and price and a chance to improve
or complete genome assemblies.

## Bioinformatics protocol

### Mapping or assembly

The first step to perform when receiving the sequencing data of your samples, is
to evaluate the sequencing quality and perform trimming and cleaning of the
reads (see [Data preprocessing](../Pipelines/data_preprocessing.md)). 

The cleaned sequence data can then be used for downstream analysis following one
of two approaches (or both in parallel, check [Data
production][../Pipelines/data_production.md]): 
- _de novo_ genome assembly of the sample(s),
- Read mapping of each sample on a reference sequence (obtained from a database
  or by _de novo_ genome assembly of one of your samples).

Both approaches are commonly used for _L. monocytogenes_.

_De novo_ genome assemblers that can be used for _L. monocytogenes_ include
[SPAdes](https://cab.spbu.ru/software/spades/) and
[Velvet](https://www.ebi.ac.uk/~zerbino/velvet/). Both of them perform very well
and are freely available. There are command-line pipelines, such as
[INNUca](https://github.com/B-UMMI/INNUca), which incorporate these programs and
provide the opportunity to automatically perform all the analyses from quality
control to genome assembly. If a platform with predefined pipelines (and that
usually does not require bioinformatics skills) is preferred, [CLC Genomic
Workbench](https://digitalinsights.qiagen.com/products-overview/discovery-insights-portfolio/analysis-and-visualization/qiagen-clc-genomics-workbench/)
and [Ridom SeqSphere+](https://www.ridom.de/seqsphere/) can be used for _L.
monocytogenes_.

As for read mapping, [BWA](https://github.com/lh3/bwa) and
[Bowtie](http://bowtie-bio.sourceforge.net/bowtie2/manual.shtml) are often used
in _L. monocytogenes_ analysis. The Center for Food Safety and Applied Nutrition
(CFSAN) of USA developed the [CFSAN
SNP](https://snp-pipeline.readthedocs.io/en/latest/) pipeline, which is tailored
to create high quality SNP matrices for sequences from closely-related
pathogens. This pipeline covers all the steps of the analysis from read mapping
to calculation of SNP distances and reconstruction of the haplotypes. Therefore,
despite requiring some bioinformatics skills, it may represent an alternative to
the development a new own pipeline and it is commonly used for _L. monocytogenes_
analysis (eg. [Hurley et al.
2019](https://msphere.asm.org/content/4/4/e00252-19), [Scaltriti et al.
2020](https://www.frontiersin.org/articles/10.3389/fpubh.2020.519293/full),
[Portmann et al.
2018)](https://www.frontiersin.org/articles/10.3389/fpubh.2020.519293/full).
Similar to the genome assembly step, a platform with predefined pipelines for
read mapping can be used. In this case, [Ridom
SeqSphere+](https://www.ridom.de/seqsphere/) is the most commonly used one. 

### Choosing a reference genome

Should an analysis require the use of a reference genome, the choice of the
reference genome is a crucial step. Analyses relying on read-mapping approaches
might be strongly influenced by reference choice, as the genetic distance
between the reference and the sample may influence the performance of downstream
steps, namely SNPs/INDELs calling ([Pightling et al.
2014](https://journals.plos.org/plosone/article?id=10.1371/journal.pone.0104579),
[Pightling et al.
2015](https://bmcresnotes.biomedcentral.com/articles/10.1186/s13104-015-1689-4)).
This reference can be picked from the samples at hand (after genome assembly),
or from a public database. If a sample is used as the reference genome, studies
on _L. monocytogenes_ usually perform preliminary analysis (e.g. hierarchical
clustering based on some distance, such as mash, ANI or allelic differences - eg.
MLST analysis) and then select a strain of each cluster to use as reference. If
instead a publicly available genome is used, the most widely used ones are [_L.
monocytogenes_ strain
EGD-e](https://www.ncbi.nlm.nih.gov/genome/159?genome_assembly_id=159660), which
is the reference genome assembly of NCBI database, [strain
CFSAN029793](https://www.ncbi.nlm.nih.gov/nuccore/NZ_CP016213) (e.g. [Ottesen et
al.
2020](https://bmcinfectdis.biomedcentral.com/track/pdf/10.1186/s12879-019-4747-z.pdf),
[Chen et al. 2017](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5293252/)) or
[strain 08-5578](https://www.ncbi.nlm.nih.gov/assembly/GCF_000093125.2), [strain
HPB5622](https://www.ncbi.nlm.nih.gov/assembly/GCF_001952775.1/) ([Pightling et
al.
2014](https://journals.plos.org/plosone/article?id=10.1371/journal.pone.0104579),
[Pightling et al.
2015](https://bmcresnotes.biomedcentral.com/articles/10.1186/s13104-015-1689-4)).
EGD-e is the reference strain for Lineage II and F2365 for reference strain
for Lineage I ([Knudsen et al.
2017](https://sfamjournals.onlinelibrary.wiley.com/doi/abs/10.1111/1758-2229.12552)).
If the objective is to discriminate between highly related samples that may/may
not belong to a single outbreak, using one outbreak isolate as reference and
combining multiple analyses approaches might maximize the resolution of your
analyses (eg. [Chen et al. 2017](https://aem.asm.org/content/83/15/e00633-17)).

### Getting SNPs

How to detect SNPs is described earlier. 

Briefly, there are three different approaches. 
- Perform _de novo_ genome assembly of each sample, and then align
  their genomic sequences. Studies involved in _L. monocytogenes_ analysis use
  very often [SPAdes](https://cab.spbu.ru/software/spades/), [CLC Genomic
  Workbench](https://digitalinsights.qiagen.com/products-overview/discovery-insights-portfolio/analysis-and-visualization/qiagen-clc-genomics-workbench/)
  or [Ridom SeqSphere+](https://www.ridom.de/seqsphere/) to obtain the assembly,
  and
  [progressiveMAUVE](http://darlinglab.org/mauve/user-guide/progressivemauve.html),
  [BLAST](https://www.ncbi.nlm.nih.gov/books/NBK279690/) or
  [MUSCLE](http://www.drive5.com/muscle/) to align the genomes. This
  multi-sequence alignment is the input for phylogenetic and clustering analysis
  (see sections on phylogeny and clustering). If instead of genome analysis,
  only the SNPs in the genes are of interest, alignments can be  performed with
  eg. [Roary](https://sanger-pathogens.github.io/Roary/) or [Panaroo](https://github.com/gtonkinhill/panaroo)
  for pangenome. 

- Use a reference genome where the reads of all the samples will be mapped, and
  then use a variant-calling pipeline to determine the polymorphic positions.
  Studies involved in _L. monocytogenes_ analysis use mostly
  [BWA](https://github.com/lh3/bwa) and
  [Bowtie](http://bowtie-bio.sourceforge.net/bowtie2/manual.shtml) for read
  mapping, and [GATK](https://gatk.broadinstitute.org/hc/en-us) and
  [VarScan](https://github.com/dkoboldt/varscan) for variant-calling. Of note,
  many of them also use the [CFSAN
  SNP](https://snp-pipeline.readthedocs.io/en/latest/) pipeline for both
  processes.

- Determine the polymorphic positions in the sample by analyzing the k-mer
  pattern using
  [kSNP](https://academic.oup.com/bioinformatics/article/31/17/2877/183216). For
  this approach either the genome assembly, or the cleaned genomic reads are
  needed. This is the less frequently used approach for _L. monocytogenes_.

Each of these approaches  will provide information about the genetic variability
in the dataset. This information can then be used to perform SNP-based
clustering and phylogenetic analysis.

### Getting alleles and allele differences

The allele sequences in the samples at hand can be retrieved by:

- Replacing the nucleotide of the reference genome by the observed alternative
  allele (check previous question), and then retrieve the sequence of each gene
  of interest considering the genome annotation of the reference.


- Obtaining the _de novo_ genome assembly of each sample, and:

  - Perform the respective genome annotation. [Prokka](https://github.com/tseemann/prokka)
    is a commonly used program
    for _L. monocytogenes_ genome annotation. [BLAST](https://blast.ncbi.nlm.nih.gov/Blast.cgi) 
    or [MUSCLE](https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-5-113)
    can be used to align
    the predicted genes to the set of genes of interest and identify homology
    relations. Alternatively, a less commonly used approach in the study of _L.
    monocytogenes_ genomes, is the use of a program like [eggNOG mapper](http://eggnog-mapper.embl.de/) 
    to perform functional annotation.
    
  - Use [BLAST](https://blast.ncbi.nlm.nih.gov/Blast.cgi) or 
    [MUSCLE](https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-5-113) 
    to align a set of genes of interest on the genome
    assembly and identify the respective homologs.
    
    
- Some allele callers, such as [ChewBBACA](https://github.com/B-UMMI/chewBBACA), 
  provide locus-specific alignments in   an automated manner, being a good option
  to determine the allelic profile of samples.

It is important to note that nowadays there are several platforms which can
automatically do all this analysis. Several of these are mentioned in the
[xMLST](../Clustering/xMLST.md) section.


### Allele-based typing

Allele-based typing consists of retrieving clustering information considering
the different alleles present in a population for a given set of genes (e.g. the
core genome). With the advent of WGS, the 7-loci based MLST approach was broadened to the use
of a cgMLST or wgMLST approach. In this context, there are two public cgMLST schemes which
have been widely used in _L. monocytogenes_ analysis considering an allele-based
approach, namely, a 1,701-loci scheme proposed by [Ruppitsch et al.
(2015)](https://pubmed.ncbi.nlm.nih.gov/26135865/), and a 1,748-loci scheme
proposed by [Moura et al.
(2016)](https://hal.archives-ouvertes.fr/pasteur-01415883). Although a
standardized approach for cgMLST analysis would be ideal, in reality both
schemes seem to work well and provide similar results ([Van Walle et al.
2018](https://www.eurosurveillance.org/content/10.2807/1560-7917.ES.2018.23.33.1700798#html_fulltext)),
and no preference is given to any of them. Nevertheless, when using automated
platforms, usually only a single scheme is available. For example,
[BIGSdb](https://pubmlst.org/software/bigsdb) uses the cgMLST scheme proposed by
[Moura et al. (2016)](https://bigsdb.pasteur.fr/listeria/), while [Ridom
SeqSphere+](https://www.ridom.de/seqsphere/) uses the scheme proposed by
[Ruppitsch et al. (2015)](http://jcm.asm.org/lookup/doi/10.1128/JCM.01193-15),
(the scheme can be found here: [https://www.cgmlst.org/ncs](https://www.cgmlst.org/ncs)). 
Besides cgMLST, some studies also perform a MVLST analysis, considering a set of virulence-related
genes, which has been shown to accurately differentiate epidemic clones (check
[Lomonaco et al. 2013](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3558006/)).

Platforms available for cgMLST typing of _L. monocytogenes_ include
[BIGSdb](https://bigsdb.pasteur.fr/listeria/) ([Jolley & Maiden
2010](https://bmcbioinformatics.biomedcentral.com/articles/10.1186/1471-2105-11-595)),
[BioNumerics](https://www.applied-maths.com/bionumerics),
[CGE](https://cge.cbs.dtu.dk/services/CGEpipeline-1.1/),
[IRIDA](https://www.irida.ca/platform-overview-2/), [Pathogen
Watch](https://pathogen.watch), and [Ridom
SeqSphere+](https://www.ridom.de/seqsphere/). However, cgMLST analysis can also
be done outside a platform with software such as
[ChewBBACA](https://github.com/B-UMMI/chewBBACA) and
[MentaLiST](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5857373/). 

### SNP-based typing

A SNP-based approach relies on the comparison of SNPs in a population. This
strategy can be seen as an alternative to the allele-based approach, but many
studies actually perform both of them and assess the overlap of the results. For
a SNP-based analysis all of the the SNPs that are present in the samples need to
be acquired and used to obtain clustering information. Examples of publicly
available pipelines for SNP-based typing are: 

- Center for Food Safety and Applied Nutrition [(CFSAN)](https://snp-pipeline.readthedocs.io/en/latest/) HqSNPs pipeline
- [Lyve-SET](https://github.com/lskatz/Lyve-SET) pipeline for HqSNPs typing 
- [SNV-Phy](https://snvphyl.readthedocs.io/en/latest/) (Canadian Public Health Agency)  
- [PHEnix](https://github.com/phe-bioinformatics/PHEnix) (The Public Health England SNP calling pipeline)

### Outbreak definition

As defined by the [World Health Organization](http://www.emro.who.int/health-topics/disease-outbreaks/index.html), “a disease outbreak is the occurrence of cases of disease in excess of what 
would normally be expected in a defined community, geographical area or season”. 
For foodborne diseases, outbreaks can be defined as two or more cases linked to the same 
food source ([Hoezler et al. 2018)](https://www.sciencedirect.com/science/article/abs/pii/S0740002017301739).
[Henri et al. (2017)](https://www.frontiersin.org/articles/10.3389/fmicb.2017.02351/full)
showed that clustering of a diverse dataset of _L. monocytogenes_ isolates
from food origin, using three different approaches (cgMLST, wgMLST and SNP based
phylogeny) was highly concordant.  

Regarding the interpretation of WGS data and the respective genetic clusters, in
the particular case of _L. monocytogenes_, some thresholds appear to be more
commonly used to define a cluster/outbreak. They are:

- ≤4 or ≤7 cgMLST allelic differences (AD) considering any of the above-mentioned
  cgMLST schemes ([Van Walle et al.
  2018](https://www.eurosurveillance.org/content/10.2807/1560-7917.ES.2018.23.33.1700798)).
  [Cabal et al.
  (2019)](https://www.frontiersin.org/articles/10.3389/fmicb.2019.02282/full)
  also considered clusters ≤ 10 AD.
  
- Using PHE [SnapperDB pipeline](https://pubmed.ncbi.nlm.nih.gov/29659710/), [Nielsen et al.
  (2017)](https://efsa.onlinelibrary.wiley.com/doi/10.2903/sp.efsa.2017.EN-1151)
  found that the maximum HqSNPs pairwise distance between outbreak isolates were
  < 5 SNPs (5/9 outbreaks) while the pairwise distances of the remaining 4/9
  outbreaks studied was between 8-21 HqSNPs. 
  
However, despite cutoff-thresholds
(eg., allelic differences for cgMLST, or SNPs pairwise differences between
isolates belonging to a single cluster) being commonly used, those thresholds
are dependent on your workflow. For example, [Chen et al
(2017)](https://aem.asm.org/content/83/15/e00633-17) demonstrated that
pairwise differences in SNP/allele count was not necessary and sufficient to
distinguish between cheese outbreak isolates from related non-outbreak
isolates. Moreover, thresholds are not directly transposable between studies,
therefore it is good practice to access the sensitivity and specificity of the
workflow when evaluating which threshold could be appropriate.

### Viruence and AMR

Several genes are important for _Listeria_ ability to cause infection and are
medically relevant, such as internalins (inlA, inlB, inlF, or inlJ, essential
for adhesion and invasion) or the prfA-regulated virulence gene cluster (pVGC)
([Vázquez-Boland et al. 2001](https://pubmed.ncbi.nlm.nih.gov/11418331/), [Ward
et al. 2004](https://pubmed.ncbi.nlm.nih.gov/15262937/), [Poimenidou et al.
2018](https://www.frontiersin.org/articles/10.3389/fmicb.2018.01103/full#B4)).
_Listeria_ are naturally susceptible to penicillin, ampicillin, amoxicillin,
gentamicin, erythromycin, tetracycline, rifampicin, co-trimoxazole, vancomycin
and imipenem ([Goméz et al.
2014](https://www.sciencedirect.com/science/article/abs/pii/S0740002014000422),
[Byrne et al.
2016](https://www.sciencedirect.com/science/article/pii/S151783821600071X#bib0255)).
Nevertheless, reports of antimicrobial resistance towards one or several of
these compounds has been reported (eg. [Boháčová et al.
2018](https://link.springer.com/article/10.1007/s12223-018-0603-6), [Escolar et
al. 2017](https://www.liebertpub.com/doi/10.1089/fpd.2016.2248), [Kevenk et al.
2015](https://onlinelibrary.wiley.com/doi/abs/10.1111/jfs.12208)). For this
reason, monitorization of virulence- and antimicrobial resistance-related genes
is of extreme relevance to determine the best way of action in presence of a
case of infection or even an outbreak. As mentioned in the [“Virulence and AMR detection"
section](../Typing/virulence_amr.md), 
where more details can be found, this is performed by comparing the genome to a database 
comprising a set of genes of interest. Examples of predefined resistome databases are 
mentioned in the same section.
