# _Salmonella_ analysis

Gram-negative bacteria of the genus _Salmonella_ are a major cause of foodborne
illness. Two _Salmonella_ species have been identified, namely, _Salmonella
enterica_ and _Salmonella bongori_. Despite only harboring two species, this genus
can be divided into several subspecies and then further to different serotypes.
Isolates are often reported by the name of the genus followed by the name of the
serotype, without mentioning the species or subspecies name ([Eng et al.
2015](https://www.tandfonline.com/doi/full/10.1080/21553769.2015.1051243)).
Moreover, _Salmonella_ isolates are usually classified into typhoidal and
non-typhoidal _Salmonella_, according to their role as causative agents of typhoid
or paratyphoid fever and salmonellosis, respectively.

_Salmonella_ serotyping is performed using the [White-Kauffman-Le
Minor](https://en.wikipedia.org/wiki/Kauffman–White_classification) scheme
([Grimont and Weill
2007](http://www.scacm.org/free/Antigenic%20Formulae%20of%20the%20Salmonella%20Serovars%202007%209th%20edition.pdf),
[Guibourdenche et al. 2010](https://pubmed.ncbi.nlm.nih.gov/19840847/)), which
uses somatic (O), flagellar (H), and capsular (Vi) antigens. This is one of the
“gold-standards” for _Salmonella_ classification, being widely used for outbreak,
surveillance and epidemiological studies. So far, more than 2,500 serotypes have
been identified, and many of them seem to be particularly associated with
certain niches
([CDC](https://www.cdc.gov/salmonella/reportspubs/salmonella-atlas/serotyping-importance.html)).
Thus, serotyping may guide public authorities during outbreak investigations.
Nevertheless, a small number of serotypes which are globally distributed are
responsible for the majority of outbreaks, and in these cases serotyping does
not have high enough resolution. Moreover, the existence of so many serotypes
obligates laboratories to keep a high amount of high-quality typing antisera and
antigens for conventional serotyping of _Salmonella_. In this context, molecular
typing methods acquired a key-role in Salmonella surveillance and outbreak
investigation.

## Typing methods

_Salmonella_ molecular typing can be performed through:

- _Pulsed Field Gel Electrophoresis (PFGE)_  
    PFGE is a fragment length restriction analysis that has long been considered
  as one of the “gold-standards” for _Salmonella_ typing, together with
  serotyping, due to its relatively high discriminatory power. This was until
  recently considered the “gold-standard” for
  [PulseNet](https://www.cdc.gov/pulsenet/index.html) network, and has been used
  by public health authorities and food regulators for outbreak investigations.


- _MLVA (Multiple locus variable tandem repeat analysis)_   
    Multiple Locus Variable Number of Tandem Repeats Analysis is a PCR-based
  typing method, which is a major typing tool used by the
  [PulseNet](https://www.cdc.gov/pulsenet/index.html) network. This method is
  able to differentiate fast-evolving bacteria even if they look similar with
  PFGE and is a faster, less laborious method. Therefore, MLVA is usually
  performed as a complement to PFGE results or instead of PFGE, thus providing a
  useful resource during outbreaks. As this analysis is specific for each
  serotype, different _Salmonella_ serotypes usually require different MLVA
  schemes. Therefore, isolates have to be serotyped before selecting the MLVA
  scheme.


- _MLST (Multi-locus Sequence Typing)_  
    As for other bacteria, a MLST method based on seven housekeeping genes (aroC,
  dnaN, hemD, hisD, thrA, sucA, and purE) has been developed for _Salmonella_
  ([Achtman et al. 2012](https://doi.org/10.1371/journal.ppat.1002776)). MLST
  can provide faster and more reproducible results compared to PFGE. However, it
  shows lower discriminatory power than PFGE and MLVA, but at a similar level as
  serotyping.


- _Microarrays_   
    The _Salmonella_ genoserotyping array (SGSA) is a microarray developed as an
  alternative to the usual serotyping method. This method presents very good
  results for the 57 most commonly reported serotypes, but fails for many
  others. Therefore, it is more useful for fast screening of those 57 serotypes,
  but not for the others. This method has been improved in [SGSA
  v2](https://pubmed.ncbi.nlm.nih.gov/30513098/).
  

- _CRISPR_   
    This method uses the diversity of spacers present at CRISPR loci to
  distinguish bacterial strains ([Fabre et al.
  2012](https://journals.plos.org/plosone/article?id=10.1371/journal.pone.0036995)).
  Amplified CRISPR loci PCR products are sequenced and analyzed to assign each
  locus to an allelic type in order to determine the allelic profile of each
  isolate, and their evolutionary relation. A CRISPR–multi-virulence-locus
  sequence typing (MVLST) approach using the genes sseL and fimH has also been
  developed ([Shariat et al. 2013](https://doi.org/10.1186/1471-2180-13-254)). A
  comparative analysis revealed that CRISPR–MVLST has a higher discriminatory
  power than the usual MLST, but lower discrimination than PFGE. This represents
  an expensive non-standardized protocol.
  

- _WGS (Whole-Genome Sequencing)_   
    With the advent of NGS technologies, WGS technology has led to the improvement
  of small salmonellosis outbreak investigation ([Kubota et al.
  2019](https://journals.sagepub.com/doi/pdf/10.1177/0033354919881650)). By
  providing information at the genomic level, WGS allows not only a highly
  discriminatory typing (cgMLST, wgMLST and SNP-typing), but also to establish
  the backward compatibility with previously mentioned molecular typing methods,
  as the molecular serotyping 7-genes MLST, which, for this reason, will tend to
  continue to be used. Furthermore, it allows the analysis of specific genes,
  such as virulence factors and antimicrobial resistance genes. Genetic
  clustering using WGS can be performed on any distance measure (eg. issued from
  allelic differences detected using cgMLST typing) or evolutionary-model based
  clustering (ie. phylogenetics) relying on variants/SNPs detection.
  [PulseNet](https://www.cdc.gov/pulsenet/index.html) network, as well as ECDC
  and EFSA, are making efforts to implement WGS as a routine tool to replace
  PFGE and MLVA. Nevertheless, in the case of _Salmonella_ this is still not a
  routine procedure.

## “One Health” surveillance and WGS of _Salmonella_

The identification of infection sources is essential for outbreak resolution.
Hence, an integrated analysis of clinical, food and veterinary samples relying
on the concept of One Health is the key to achieve a good surveillance system.
As shown
[here](https://www.cdc.gov/listeria/surveillance/whole-genome-sequencing.html)
by PulseNet network for _Listeria_, the high discriminatory power of WGS increases
the chances to find the bacterial source of infection, and possibly reduces the
time that it takes. Indeed, as reported by the
[WHO](https://apps.who.int/iris/bitstream/handle/10665/272430/9789241513869-eng.pdf?sequence=1),
the use of WGS increased the resolution of _Salmonella_ cluster analysis, and
contributed to the identification of recurrent sources of infection.
Furthermore, the integrated WGS analysis of food and human samples at
international level during a multi-country _Salmonella_ outbreak allowed the
identification of the source of infection in Germany ([Inns et al.
2015](https://pubmed.ncbi.nlm.nih.gov/25953273/)), reflecting the ease at which
WGS data can be shared, analyzed and compared. However, several factors are
hindering the implementation of a generalized WGS-based surveillance system.
For instance, the resources for implementation of WGS differ between different
sectors (human health, animal health and food safety), thus complicating the
implementation of “One health” surveillance. For this reason, it has been
decided that the technological transition to WGS-based surveillance at European
level is performed first in _Listeria_, and only afterwards in other bacteria,
such as _Salmonella_ ([ECDC
roadmap](https://www.ecdc.europa.eu/en/publications-data/ecdc-roadmap-integration-molecular-typing-and-genomic-typing-european-level)).

## WGS lab protocol

### DNA extraction

Regarding DNA extraction, there is not a standard protocol or kit that is used,
but a protocol directed towards Gram-negative bacteria will be recommended.


### Sequencing technology

There is not a prefered WGS technology to sequence _Salmonella_. Similar to other
fields, Illumina paired-end reads represent the most commonly used strategy. Due
to the number of samples that can be handled at a single run and the possible
higher read size, MiSeq sequencing machines seem to be the choice for the
majority of the labs. Long-read sequencing technologies are now becoming more
frequently used, and there is an apparent tendency to sequence _Salmonella_
genomes using both short- and long-read technologies. 

For Illumina sequencing, the choice of library preparation procedure may have
adverse effects on in silico serotyping. For example, the Nextera XT library
preparation kit seems to introduce a GC bias, which negatively affects O-antigen
recognition due to increased fragmentation ([Uelze,
2019](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7028957/)). The new version,
Nextera Flex, is therefore recommended over the XT kit.

## Bioinformatics protocol

### Mapping or assembly

The first step to perform when receiving the sequencing data of your samples, 
is to evaluate the sequencing quality and perform trimming and cleaning of the reads. 

The cleaned sequence data can then be used for downstream analysis following one
of two approaches (or both in parallel, check [Data
production](../Pipelines/data_production.md)): 
- _De novo_ genome assembly of the sample(s),
- Read mapping of each sample on a reference sequence (obtained from a database
  or by de novo genome assembly of one of your sample)

It is important to note that both approaches have advantages and disadvantages,
and the decision on which of them to follow should be made according to what is
more appropriate for the data you have at hand, and the purpose of your
analyses. De novo genome assembly of all sequenced isolates followed by their
annotation seems to be a common approach in studies including Salmonella
genomes. Nevertheless, especially when a further SNP-based approach will be
performed (see next questions), a parallel read mapping approach is also
followed.

_De novo_ genome assemblers that can be used for _Salmonella_ include
[SPAdes](https://cab.spbu.ru/software/spades/) and
[SKESA](https://genomebiology.biomedcentral.com/articles/10.1186/s13059-018-1540-z).
Both of them perform very well and are freely available. A major difference
between them is the fact that SKESA can not use longreads produced by Oxford
Nanopore or Pacfic Biosciences machines. For this reason, it does not represent
a good alternative for hybrid assemblies combining both short- and long-reads,
which is the tendency in the field. In this context,
[Unicycler](https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1005595),
a pipeline tailored to perform hybrid assemblies, combines SPAdes to other
tools,  is commonly used for _Salmonella_ genomes. Other command-line pipelines,
such as [INNUca](https://github.com/B-UMMI/INNUca), also provide the opportunity to automatically perform all the
analyses from quality control to genome assembly. If a platform with predefined
pipelines is needed instead,
[INNUENDO](https://www.efsa.europa.eu/en/supporting/pub/en-1498),
[BioNumerics](https://www.applied-maths.com/bionumerics), [Ridom
SeqSphere+](http://www3.ridom.de/seqsphere/) and
[EnteroBase](https://enterobase.readthedocs.io/en/latest/pipelines/backend-pipeline-qassembly.html)
can be used for _Salmonella_.

As for read mapping, [BWA](http://bio-bwa.sourceforge.net) is a common choice
for _Salmonella_. Alternatively, [CFSAN
SNP](https://snp-pipeline.readthedocs.io/en/latest/) pipeline, which is tailored
to create high quality SNP matrices for sequences from closely-related
pathogens, is a commonly used pipeline for _Salmonella_. This pipeline covers all
the steps of the analysis from read mapping to calculation of SNP distances and
reconstruction of the haplotypes. Therefore, despite requiring some
bioinformatics skills, it may represent a good alternative to the development of
your own pipeline. Noteworthy, as mentioned before, these represent commonly
used approaches, and not recommendations. Thus, other methodologies, pipelines
or even platforms may be used for your analysis.


### Choosing a reference genome

Should the analysis require the use of a reference genome, the choice of the
reference genome is a crucial step. Analyses relying on read-mapping approaches
might be strongly influenced by reference choice, as the genetic distance
between the reference and the sample may influence the performance of downstream
steps, namely SNPs/INDELs calling ([Pightling et al.
2014](https://journals.plos.org/plosone/article?id=10.1371/journal.pone.0104579),
[Pightling et al.
2015](https://bmcresnotes.biomedcentral.com/articles/10.1186/s13104-015-1689-4)).
This reference can be picked from the samples themselves (after genome
assembly), or from a public database. In both cases the reference must be chosen
according to the serotype of each isolate. For this reason, it is essential to
determine the serotype before read mapping, and there is not a specific
reference genome that is used from public databases. However, a closed bacterial
genome will be preferable.

### _Salmonella_ serotyping

As mentioned before, determination of _Salmonella_ serotype is an important step
to be able to perform further analysis. For instance, a read-mapping approach
and downstream analysis obtain better results if the reference genome
corresponds to the same serotype as the sample. Serotype determination can be
performed with the [White-Kauffman-Le
Minor](https://en.wikipedia.org/wiki/Kauffman–White_classification) scheme, or
with an _in silico_ pipeline. The most commonly used programs for _in silico_
serotype determination in _Salmonella_ are
[SISTR](https://lfz.corefacility.ca/sistr-app/) and
[SeqSero](https://github.com/denglab/SeqSero) (and v2.
[SeqSero2](https://aem.asm.org/content/85/23/e01746-19)). Although less commonly
used, the [“bacterial analysis
pipeline”](https://cge.cbs.dtu.dk/services/cge/abstract.php) is also an option.
The [Salmonella Type
Finder](https://cge.cbs.dtu.dk/services/SalmonellaTypeFinder/) is a pipeline
developed by the Center for Genomic Epidemiology which uses
[SRST2](https://github.com/katholt/srst2) and
[SeqSero](https://github.com/denglab/SeqSero). Enterobase is a complete pipeline
combining several tools and includes both SISTR and SeqSero2.

### Getting SNPs

How to detect SNPs is described earlier. Briefly, there are three different
approaches.

- Perform _de novo_ genome assembly of each sample (check above), and then align
  their genomic sequences. _Salmonella_ analyses usually use
  [MAUVE](http://darlinglab.org/mauve/mauve.html) to align the genomes. This
  multi-sequence alignment can be then input to
  [SNP-sites](https://github.com/sanger-pathogens/snp-sites) to get the number
  of variants.

- Use a reference genome where the reads of all the samples will be mapped, and
  then use a variant-calling pipeline to determine the polymorphic positions.
  [CFSAN SNP](https://snp-pipeline.readthedocs.io/en/latest/) is a commonly used
  pipeline which performs both processes (read mapping and variant calling).
  [Snippy](https://github.com/tseemann/snippy) and
  [SNVPhyl](https://snvphyl.readthedocs.io/en/latest/user/input/#input) are also
  commonly used alternatives for _Salmonella_ genomes.

- Determine the polymorphic positions in the sample by analyzing the k-mer
  pattern using
  [kSNP](https://academic.oup.com/bioinformatics/article/31/17/2877/183216). For
  this approach you can either provide the genome assembly, or the cleaned
  genomic reads. This is the less frequently used approach for _Salmonella_.

Each of these approaches provides you with information about the genetic
variability in the dataset. This information can then be used to perform
SNP-based clustering and phylogenetic analysis. Alternatively, if a read
mapping approache is followed, the reference nucleotide can be replaced by the
observed allele, and consequently reconstruct the haplotype of each sample. This
is the approach used by the [CFSAN SNP
pipeline](https://snp-pipeline.readthedocs.io/en/latest/).


### Getting alleles and allele differences

The allele sequences of the samples can be retrieved by:

- Replacing the nucleotide of the reference genome by the observed alternative
  allele (check previous question), and then retrieve the sequence of each gene
  of interest considering the genome annotation of the reference.
- Obtaining the _de novo_ genome assembly of each sample, and performing the
  respective genome annotation. [Prokka](https://github.com/tseemann/prokka) and
  [NCBI Prokaryotic Genome Annotation
  Pipeline](https://www.ncbi.nlm.nih.gov/genome/annotation_prok/) are commonly
  used programs for _Salmonella_ genome annotation.
  [GLIMMER](http://ccb.jhu.edu/software/glimmer/index.shtml) and
  [RASTk](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4322359/) are also used.
- Some allele callers, such as [chewBBACA](https://github.com/B-UMMI/chewBBACA),
  provide locus-specific alignments in an automated manner, being a good option
  to determine the allelic profile of  samples.

It is important to note that nowadays there are several platforms which can
automatically do all this analysis. One of the more commonly used for _Salmonella_
is [Enterobase](https://enterobase.readthedocs.io/en/latest/index.html), which
provides assembly, serotyping and allele calling. Several of these platforms are
mentioned in the [xMLST](../Clustering/xMLST.md) section.

### Allele based typing

Allele-based typing consists of retrieving clustering information considering
the different alleles present in a population for a given set of genes (e.g. the
core genome). With the developmentadvent of WGS, the 7-loci based MLST approach was broadened
to the use of a cgMLST approach. In this context, there is a public cgMLST
scheme which has been widely used in _Salmonella enterica_ analysis. This scheme
comprises [3,002 loci](https://www.cgmlst.org/ncs/schema/4792159/), and is
available in the most commonly used platforms, such as
[EnteroBase](https://enterobase.readthedocs.io/en/latest/pipelines/salmonella-statistics.html)
and [Ridom SeqSphere+](https://www.ridom.de/seqsphere/).
[EnteroBase](https://enterobase.readthedocs.io/en/latest/pipelines/salmonella-statistics.html)
and
[BioNumerics](http://www.applied-maths.com/sites/default/files/extra/Release-Note-Salmonella-enterica-schema.pdf)
also use a wgMLST scheme which, besides the previously mentioned cgMLST loci,
include the accessory genes.

Platforms available for cgMLST typing of _Salmonella_ include
[Enterobase](https://enterobase.readthedocs.io/en/latest/index.html),
[INNUENDO](https://www.efsa.europa.eu/en/supporting/pub/en-1498),
[BioNumerics](https://www.applied-maths.com/bionumerics),
[CGE](https://cge.cbs.dtu.dk/services/CGEpipeline-1.1/),
[IRIDA](https://www.irida.ca/platform-overview-2/), [Pathogen
Watch](https://pathogen.watch), and [Ridom
SeqSphere+](https://www.ridom.de/seqsphere/).

### SNP based typing

A SNP-based approach relies on the comparison of SNPs in a population. This
strategy can be seen as an alternative to the allele-based approach, but many
studies actually perform both of them and assess the overlap of the results. For
a SNP-based analysis all of the the SNPs that are present in the samples need to
be acquired and used to obtain clustering information. Examples of publicly
available pipelines for SNP-based typing are: 

- Center for Food Safety and Applied Nutrition [(CFSAN)](https://snp-pipeline.readthedocs.io/en/latest/) HqSNPs pipeline
- [Lyve-SET](https://github.com/lskatz/Lyve-SET) pipeline for HqSNPs typing 
- [SNV-Phy](https://snvphyl.readthedocs.io/en/latest/) (Canadian Public Health Agency)  
- [PHEnix](https://github.com/phe-bioinformatics/PHEnix) (The Public Health England SNP calling pipeline)

### Outbreak definition

As defined by the [World Health Organization](http://www.emro.who.int/health-topics/disease-outbreaks/index.html), 
“a disease outbreak is the
occurrence of cases of disease in excess of what would normally be expected in a
defined community, geographical area or season”. WGS data provides a high
discriminatory power allowing clustering of different isolates (from different
geographical areas, and clinical, animal or environmental sources) according to
their genomic similarity. This contributes not only to an earlier detection of
outbreaks and determination of contamination sources, but also to the detection
of more outbreaks, as has been reported by [PulseNet](https://www.cdc.gov/listeria/surveillance/whole-genome-sequencing.html) 
network for _Listeria_. Nevertheless, it is still
difficult to establish a clear cluster outbreak definition for _Salmonella_, a threshold at which
we decide whether two isolates belong to the same genetic cluster, thus linking
two cases of infection. 

### Virulence and AMR

Several genes are important for _Salmonella_ ability to cause infection and are
medically relevant, such as motility genes, fimbrial adhesins and metabolic
genes ([Ilyas et al.
2017](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5626846/), [Eng et al.
2015](https://www.tandfonline.com/doi/full/10.1080/21553769.2015.1051243)). In
the particular case of _Salmonella_, horizontally transferred genes strongly
influence the course of infection, as they can lead to the emergence of new
phenotypes and favor the adaptation to new niches ([Ilyas et al.
2017](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5626846/), [Ochman et al.
2000](https://pubmed.ncbi.nlm.nih.gov/10830951/)). Such events are not only
important for _Salmonella_ ability to infect humans, but also for the acquisition
of resistance to antimicrobial drugs ([Wang et al.
2019](https://www.frontiersin.org/articles/10.3389/fmicb.2019.00985/full#B65)).
Chloramphenicol, ampicillin, and trimethoprim–sulfamethoxazole are the
first-line antimicrobial drugs used to treat _Salmonella_ infections. However,
over the years resistance towards one or several of these drugs (leading to
multidrug resistant isolates) has been emerging ([Eng et al.
2015](https://www.tandfonline.com/doi/full/10.1080/21553769.2015.1051243)).
Alternative antimicrobials have been used. However, resistance towards the
alternatives is also appearing, and antimicrobial resistance in _Salmonella_ is
considered a global threat ([Marchello et al.
2020](https://www.ajtmh.org/view/journals/tpmd/103/6/article-p2518.xml), [Eng et
al. 2015](https://www.tandfonline.com/doi/full/10.1080/21553769.2015.1051243)).
For this reason, monitoring of virulence- and antimicrobial resistance-related
genes is of great relevance to determine the best way of action in the presence
of a case of infection or even an outbreak. As mentioned in the [Virulome and
Resistome section](../Typing/virulence_amr.md), where more details can be found, this is performed by
comparing the genome to a database comprising a set of genes of interest.
Examples of predefined resistome databases are mentioned in the same section.
